staging/src/k8s.io/client-go/util/jsonpath/jsonpath.go

```diff
@@ -297,11 +297,11 @@ func (j *JSONPath) findFieldInValue(value *reflect.Value, node *FieldNode) (refl
 	var inlineValue *reflect.Value
 	for ix := 0; ix < t.NumField(); ix++ {
 		f := t.Field(ix)
-		jsonTag := f.Tag.Get("json")
-		parts := strings.Split(jsonTag, ",")
-		if len(parts) == 0 {
+		jsonTag, found := f.Tag.Lookup("json")
```

Comment:
hmm, Go's logic for JSON-encoding structs with anonymous fields is substantially more complex than I had realised.
https://github.com/golang/go/blob/master/src/encoding/json/encode.go#L111-L130
https://github.com/golang/go/blob/master/src/encoding/json/encode.go#L1065-L1282

I don't yet fully understand how kubernetes actually serialises objects, whether it uses Go's json library at all (e.g. its usage of the "inline" tag).

However, I fear that any logic in jsonpath that doesn't exactly match the json encoder that kubernetes uses must probably be wrong (i.e. taking the "short cut" of running a jsonpath query against the underlying object could return a different result than running the query against the object's serialisation).

I think that the path forward would probably be to bring the relevant functions that the json encoder uses (wherever those are) into jsonpath.
Or for jsonpath to work on json serialised data rather than the deserialised objects.